dht/addr.go

```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // Used internally to refer to node network addresses.
-type dHTAddr interface {
+type DHTAddr interface {
```

Comment:
Fixed! Now casts to net.Addr prior to passing to hooks. I'm not as familiar with the *.Addr types as you, so I'll take it on faith for now that casting is safe. :)